src/routes/profile-route/profile-settings-route.jsx

```diff
@@ -0,0 +1,205 @@
+    setValue,
+    getValues,
+  } = useForm({
+    defaultValues: {
+      username: "",
+      email: "",
+      password: "",
+    },
+  });
+
+  const onSubmit = () => {
+    //loading should be true
+    //get values from form
+    const { username, email, password } = getValues();
+    //make sure that new username is unique
+    //const usersQuery = useQuery(GET_USERS, { variables: { where: {} } });
+    //usersQuery.map((username) => {
+    //});
+
+    updateUser({
+      variables: {
+        where: { id: userId },
+        data: { username, email, password },
+      },
+    })
+      .then((res) => {
+        console.log(res);
+        //form data is not currently refreshing after successful update
+        //loading(false);
+        //probably need to change button text to "Changes Saved" after submit?
+      })
+      .catch(error);
+  };
+
+  const isDesktopSize = useMediaQuery({
+    query: "(min-width: 1170px)",
+  });
+
+  return (
+    <div className="max-w-[1440px] mx-auto">
+      <Header>
+        <Header.Logo />
+        <Header.NavLinks />
+        <Header.Profile />
+      </Header>
+
+      <section className="flex flex-colmax-w-[1440px] h-[900px] bg-gray-background">
+        {isDesktopSize ? <ProfileSidebar /> : <ProfileHorizontalMenu />}
+
+        <section className="mt-9">
+          <Form onSubmit={handleSubmit(onSubmit)}>
+            <Form.TextInput
+              register={register("username", {
+                required: "Nickname is required",
+                pattern: {
+                  value: /^\S/,
+                  message: "Must not start with a space",
+                },
+              })}
+              labelText={"Nickname"}
+              onChange={(evt) => {
+                setValue("username", evt.target.value, {
+                  shouldValidate: true,
+                });
+              }}
+              isValid={!errors.username}
+              placeholder="Type here"
+              prefilledInputs={prefilledInputs.username}
+            />
+
+            {errors.username && (
+              <Form.Feedback message={errors.username.message} />
+            )}
+
+            <Form.TextInput
+              register={register("email", {
+                required: "Email is required",
+                pattern: {
+                  value: /[\w\-\.]+@([\w\-]+\.)+[\w\-]{2,4}/,
+                  message: "Invalid Email",
+                },
+              })}
+              labelText={"Email"}
+              onChange={(evt) => {
+                setValue("email", evt.target.value, {
+                  shouldValidate: true,
+                });
+              }}
+              isValid={!errors.email}
+              placeholder="Type here"
+              prefilledInputs={prefilledInputs.email}
+            />
+
+            {errors.email && <Form.Feedback message={errors.email.message} />}
+
+            <Form.TextInput
+              register={register("password", {
```

Comment:
Don't know if passwords should be required right now. Works perfectly fine as is, but seems like they are there for confirming the changes rather than changing the password since regardless of if the values are going to change or not, a value is still required to submit. Also could cause issues because if a user does not know that the value will change in the server, they could change their password without realizing it. Might combine the above comment so all values are not required to change or have a value for the form to submit. (ex: if(username || email || password) change then ...)